web/app/js/components/NavigationResource.jsx

```diff
@@ -0,0 +1,145 @@
+import {
+  Collapse,
+  ListItemIcon,
+  ListItemText,
+  MenuItem,
+  MenuList,
+} from '@material-ui/core';
+
+import ExpandMore from '@material-ui/icons/ExpandMore';
+import { Link } from 'react-router-dom';
+import NavigateNextIcon from '@material-ui/icons/NavigateNext';
+import PropTypes from 'prop-types';
+import React from 'react';
+import SuccessRateDot from "./util/SuccessRateDot.jsx";
+import _ from 'lodash';
+import { friendlyTitle } from "./util/Utils.js";
+import { processedMetricsPropType } from './util/MetricUtils.jsx';
+import { withContext } from './util/AppContext.jsx';
+import { withStyles } from '@material-ui/core/styles';
+
+const styles = () => ({
+  navMenuItem: {
+    paddingLeft: "24px",
+    paddingRight: "24px",
+  },
+  navResourceItem: {
+    marginRight: "0",
+    paddingLeft: "9px",
+  },
+  navResourceText: {
+    textOverflow: "ellipsis",
+    overflow: "hidden",
+  }
+});
+
+class NavigationResource extends React.Component {
+  static defaultProps = {
+    metrics: [],
+  }
+
+  static propTypes = {
+    api: PropTypes.shape({
+      prefixLink: PropTypes.func.isRequired,
+    }).isRequired,
+    classes: PropTypes.shape({}).isRequired,
+    metrics: PropTypes.arrayOf(processedMetricsPropType),
+    type: PropTypes.string.isRequired,
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.resources = _(props.metrics)
+      .filter(m => m.pods.meshedPods !== "0")
+      .map(m =>
+        _.merge(m, {
+          menuName: props.type === "namespaces" ? m.name : `${m.namespace}/${m.name}`
+        })
+      )
+      .sortBy("menuName")
+      .value();
+    this.to = props.api.prefixLink("/" + props.type);
+
+    this.state = {open: false};
+  }
+
+  handleOnClick = () => {
+    if (!_.isEmpty(this.props.metrics)) {
+      // not expandable if no metrics
+      this.setState({ open: !this.state.open });
+    }
+  };
+
+
+  listItemIcon() {
+    let icon = <NavigateNextIcon color="disabled" />;
+    if (!_.isEmpty(this.resources)) {
+      if (this.state.open) {
+        icon = <ExpandMore />;
+      } else {
+        icon = <NavigateNextIcon />;
+      }
+    }
+    return (<ListItemIcon>{icon}</ListItemIcon>);
+  }
+
+  menu() {
+    const { classes, type } = this.props;
+
+    return (
+      <MenuItem
+        component={Link}
+        to={this.to}
+        className={classes.navMenuItem}
+        onClick={this.handleOnClick}
+        selected={this.to === window.location.pathname}>
+        {this.listItemIcon(this.resources)}
+        <ListItemText primary={friendlyTitle(type).plural} />
+      </MenuItem>
+    );
+  }
+
+  subMenu() {
+    const { api, classes } = this.props;
+
+    return (
+      <MenuList dense component="div" disablePadding>
```

Comment:
what do you think about making the first subMenu Item "All Pods" (etc), rather than navigating to the pods list page  when you click to expand the submenu? I find it a little disconcerting if I'm, for example, in the middle of a top page, and I just want to see what resources are available in the sidebar, but I get navigated off the current page.

Reply:
@rmars agree it's disorienting when navigating the resource tree causes the page to change. in the second commit, i tried adding a small button, visible on hover, that provides the "all" link you described, while keeping the rest of the menu item clickable for navigation only. let me know what you think.

Reply:
hmmm do you have objections to just adding an "All <resource>s" option as the first option in the list? I think it's weird to have another icon just pop up that you don't know what it does (even if it's the list icon that Resources uses). I feel like having an "All <resource>" option makes it very clear as to what is contained in the option. 

Having options to click an adjacent icon as well as click a list item also feels weird in that it divides the actions you could take. This introduces more complication than the simple "I click on a menu and then a submenu" flow. What would be the motivations for breaking this flow?

Reply:
@rmars Added an "All" list item as you suggested. You're right a random button appearing is a bit confusing. I was trying to avoid having a list item at the top of the menu that's different from the individual resources below it, as I haven't seen that pattern in a tree navigation UI before. That said, this makes the code way simpler, and is the most explicit for the user. Thanks for the thorough eyes!

Reply:
yeah, I do wonder how most people actually do this! thanks for adding!

Reply:
ooh, thanks for finding these! pop-up style is pretty cool. something to consider down the road.